projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -329,16 +329,31 @@ public function render_button( $attributes, $content = null, $block = null ) {
 	public function get_subscription_url( $plan_id ) {
 		global $wp;
 
-		return add_query_arg(
-			array(
-				'blog'     => esc_attr( self::get_blog_id() ),
-				'plan'     => esc_attr( $plan_id ),
-				'lang'     => esc_attr( get_locale() ),
-				'pid'      => esc_attr( get_the_ID() ), // Needed for analytics purposes.
-				'redirect' => esc_attr( rawurlencode( home_url( $wp->request ) ) ), // Needed for redirect back in case of redirect-based flow.
-			),
-			'https://subscribe.wordpress.com/memberships/'
+		$query_args = array(
+			'blog'     => esc_attr( self::get_blog_id() ),
+			'plan'     => esc_attr( $plan_id ),
+			'lang'     => esc_attr( get_locale() ),
+			'pid'      => esc_attr( get_the_ID() ), // Needed for analytics purposes.
+			'redirect' => esc_attr( rawurlencode( home_url( $wp->request ) ) ), // Needed for redirect back in case of redirect-based flow.
 		);
+
+		if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+			return add_query_arg( $query_args, 'https://subscribe.wordpress.com/memberships/' );
```

Comment:
I'm not sure I follow. Shouldn't we pass the customAmount parameter for jetpack/Atomic sites too? 🤔

Reply:
Oh my, I tested on a Jurassic site using a non-donation version payment button which does always show the custom amount...

It seems like I'm going to need to another approach for this.